tests/app/core/models/session-model.spec.ts

```diff
@@ -20,67 +20,67 @@ describe('Session Model', () => {
         });
     });
 
-    it('should throw error on session data with missing crucial values', () => {
-      of({ dummyKey: undefined })
+    it('should throw error on session data with invalid session', () => {
+      of({ ...VALID_SESSION_DATA, sessionRepo: null })
         .pipe(assertSessionDataIntegrity())
         .subscribe({
           next: () => fail(),
-          error: (err) => expect(err).toEqual(new Error(SESSION_DATA_MISSING_OPENPHASES_KEY))
+          error: (err) => expect(err).toEqual(new Error(SESSION_DATA_MISSING_FIELDS))
         });
-    });
-
-    it('should throw error on session with no open phases', () => {
-      of(NO_OPEN_PHASES_SESSION_DATA)
+      of({ ...VALID_SESSION_DATA, sessionRepo: [] })
+        .pipe(assertSessionDataIntegrity())
+        .subscribe({
+          next: () => fail(),
+          error: (err) => expect(err).toEqual(new Error(SESSION_DATA_MISSING_FIELDS))
+        });
+      of({ ...VALID_SESSION_DATA, sessionRepo: 'repo' })
         .pipe(assertSessionDataIntegrity())
         .subscribe({
           next: () => fail(),
-          error: (err) => expect(err).toEqual(new Error(NO_ACCESSIBLE_PHASES))
+          error: (err) => expect(err).toEqual(new Error(SESSION_DATA_MISSING_FIELDS))
         });
     });
 
-    it('should throw error on session data with invalid open phases', () => {
-      of({ ...BUG_REPORTING_PHASE_SESSION_DATA, openPhases: ['unknownPhase'] })
+    it('should throw error on session with invalid phases', () => {
+      of({ ...VALID_SESSION_DATA, sessionRepo: [{ phase: 'invalidPhase' as Phase, repos: [WATCHER_REPO] }] })
         .pipe(assertSessionDataIntegrity())
         .subscribe({
           next: () => fail(),
           error: (err) => expect(err).toEqual(new Error(NO_VALID_OPEN_PHASES))
         });
     });
 
-    it('should throw error on session data with undefined repo for open phase', () => {
-      of({ ...BUG_REPORTING_PHASE_SESSION_DATA, phaseBugReporting: undefined })
+    it('should throw error on session data with invalid repo', () => {
+      of({ ...VALID_SESSION_DATA, sessionRepo: [{ phase: Phase.issuesViewer, repo: undefined }] })
```

Comment:
This might be a good place to use spread syntax to override object properties but probably something like this
```
of({ sessionRepo: [{ ...ISSUES_VIEWER_SESSION_REPO, repo: undefined }] })
```
but I'll also be fine if you just do
```
of({ sessionRepo: [{ phase: Phase.issuesViewer, repo: undefined}] })
```